web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,132 @@
+      const endTime: moment.Moment = moment.utc(
+        `${encodedDate} ${encodedTime}`,
+        'DDMMYYYY hh mm ss.SS'
+      )
+      return endTime
+    }
+  }
+  return moment.utc('')
+}
 
-const parseDateFromLogFileName = (fileName: string): string => {
-  const fileNameArr = fileName.split('_')
-  const dateString = fileNameArr[1]
-  return `${dateString.slice(0, 4)}/${dateString.slice(
-    4,
-    6
-  )}/${dateString.slice(6)}`
+export const getDustrakStart = (textLines: Array<string>): moment.Moment => {
+  try {
+    const startTime: string = textLines[6].split(',')[1]
+    const startDate: string = textLines[7].split(',')[1]
+    const startDatetime: moment.Moment = moment.tz(
+      `${startDate} ${startTime}`,
+      'MM-DD-YYYY hh:mm:ss a',
+      'America/Los_Angeles'
+    )
+    return startDatetime
+  } catch (e) {
+    if (e instanceof TypeError) {
+      console.warn(e)
+      return moment('')
+    } else {
+      throw e
+    }
+  }
 }
 
-export const validateFiles = (files: Array<FileWithPath>): string => {
-  if (files.length > 0 && files.length < 2) {
-    return 'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
+export const getDustrakEnd = (
+  textLines: Array<string>,
+  startDatetime: moment.Moment
+): moment.Moment => {
+  try {
+    const endDatetime: moment.Moment = startDatetime.clone()
+    const testLength: string = textLines[8].split(',')[1]
+    const [days, hours, minutes]: Array<string> = testLength.split(':')
+    endDatetime
+      .add(days, 'days')
+      .add(hours, 'hours')
+      .add(minutes, 'minutes')
+    return endDatetime
+  } catch(e){
+    if (e instanceof TypeError){
+      console.warn(e)
+      return moment('')
+    } else{
+      throw e
+    }
   }
+}
 
-  if (files.length > 2) {
-    return 'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
+export const validateFiles = async (
+  files: Array<FileWithPath>
+): Promise<ValidateMeta> => {
+  const dustrakData: DustrakMeta = {
+    startDatetime: '',
+    endDatetime: ''
+  }
+  const validateData: ValidateMeta = {
+    message: '',
+    dustrakMeta: dustrakData
   }
 
-  if (files.length === 2) {
-    let logFile: FileWithPath | undefined
-    let csvFile: FileWithPath | undefined
+  if (files.length > 0 && files.length < 2) {
+    validateData.message =
+      'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
+  } else if (files.length > 2) {
+    validateData.message =
+      'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
+  } else if (files.length === 2) {
+    let logFile: File | undefined
+    let csvFile: File | undefined
 
     files.forEach(file => {
-      if (file.name.endsWith('.csv')) {
+      if (file.type === 'text/csv') {
         csvFile = file
       }
 
-      if (file.name.endsWith('.log')) {
+      if (file.type === 'text/x-log') {
```

Comment:
I'm on a mac and the `file.type` for my .log file shows up as an empty string for some reason:
![image](https://user-images.githubusercontent.com/18567829/81761619-3938c280-947f-11ea-802f-39e7e0694965.png)

As a result, I failed this validation with these two files: https://drive.google.com/drive/folders/1_cbfTu6-GiCgZzez7TCBef4dVv0HK2m3

When I previously researched (briefly) checking for file type, I think there are some inconsistencies when it comes to these checks, and we might only be able to check for file extensions.

Reply:
**Response Action**
I'll revert
**Context**
Ugh, cross-system compatibility.

Reply:
I happened upon this discussion last week: https://github.com/react-dropzone/react-dropzone/issues/276
It seems that MIME types are extremely unreliable across browsers and operating systems, supporting your conclusion that extensions sound like the best way to go.